api/data_explorer/util/elasticsearch_util.py

```diff
@@ -215,13 +251,39 @@ def get_elasticsearch_facet(es, elasticsearch_field_name, field_type):
             field=elasticsearch_field_name,
             interval=_get_bucket_interval(field_range))
 
-    # Handle sample facets in a special way since they are nested objects.
-    if elasticsearch_field_name.startswith('samples.'):
-        es_facet = NestedFacet('samples', es_facet)
-
+    nested_facet = _get_nested_facet(elasticsearch_field_name, es_facet,
+                                     nested_paths)
+    if nested_facet:
+        es_facet = nested_facet
     return es_facet
 
 
+def _get_nested_facet(elasticsearch_field_name, es_facet, nested_paths):
+    """
+    Returns a NestedFacet for the elasticsearch field, if the field is nested.
+    """
+    parent = elasticsearch_field_name.rsplit('.', 1)[0]
+    nested_facet = None
+    is_nested = True
```

Comment:
This is confusing, elasticsearch_field_name may not be nested.

Can this method just be:
```
# Say nested_paths is: ["outer", "outer.inner"] and elasticsearch_field_name is "outer.inner.field".
# Traverse nested_paths in reverse order to make sure we use "outer.inner" instead of "outer".
for path in nested_paths in backwards order:
  if elasticsearch_field_name = '.' in path:
    return NestedFacet
```

If it is this simple, it doesn't need to be a separate method -- up to you.